src/amber.cc

```diff
@@ -54,15 +54,7 @@ amber::Result Amber::Execute(const std::string& input, const Options& opts) {
   if (!engine)
     return Result("Failed to create engine");
 
-  if (opts.default_device)
-    r = engine->InitializeWithDevice(opts.default_device);
-  else
-    r = engine->Initialize();
-
-  if (!r.IsSuccess())
-    return r;
-
-  r = executor->Execute(engine.get(), parser->GetScript());
+  r = executor->Execute(engine.get(), parser->GetScript(), opts.default_device);
```

Comment:
I'd like to avoid combing initialize and execute. It makes things a lot more intertwined.